packages/core/src/browser/preferences/preference-proxy.spec.ts

```diff
@@ -268,8 +270,8 @@ describe('Preference Proxy', () => {
                     }
                 });
                 await prefService.set('my.pref', 'bog', PreferenceScope.User);
-                expect(changesNotAffectingTypescript, 'Two events (one for `my.pref` and one for `[swift].my.pref`) should not have affected TS').to.equal(2);
-                expect(changesAffectingTypescript, 'One event should have been fired that does affect typescript.').to.equal(1);
+                expect(changesNotAffectingTypescript, 'Four events (two for `my.pref` and two for `[swift].my.pref`) should not have affected TS').to.equal(4);
+                expect(changesAffectingTypescript, 'Two events should have been fired that do affect typescript.').to.equal(2);
```

Comment:
I think that this is precisely the outcome that these tests were designed to prevent.

Rather than unconditionally allowing changes through, I believe that some logic for merging object-typed preferences should be used to ensure that no more events are fired than necessary, but their content reflects the merged state.